On branch ControlNet-v1-1

GoooHi opened the pull request:
1. 新增权重 control_v11e_sd15_ip2p 和 对应文件 gradio_ip2p2image.py
2. 新增权重 control_v11e_sd15_shuffle 和 对应文件 gradio_shuffle2image.py

合并代码至https://github.com/PaddlePaddle/PaddleNLP/tree/36591ab297dfa45d89255aa6e14988940ccea299/ppdiffusers/examples/controlnet

GoooHi committed:
On branch ControlNet-v1-1
Changes to be committed:
	new file:   ../../ppdiffusers/examples/controlnet/gradio_ip2p2image.py
	new file:   ../../ppdiffusers/examples/controlnet/gradio_mlsd2image.py

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PaddlePaddle/PaddleNLP?pullRequest=5951) <br/>All committers have signed the CLA.

JunnYu commented on ppdiffusers/examples/controlnet/gradio_mlsd2image.py:
```diff
@@ -0,0 +1,134 @@
+
+apply_mlsd = MLSDdetector()
+
+controlnet = ControlNetModel.from_pretrained("F:/Paddle/model/control_v11p_sd15_mlsd")
```
这里可以 使用 https://huggingface.co/lllyasviel/control_v11p_sd15_mlsd 的名字 "lllyasviel/control_v11p_sd15_mlsd"

Submerge-Gu replied:
好的，那我现在是修改后重新提交吗

JunnYu commented on ppdiffusers/examples/controlnet/gradio_ip2p2image.py:
```diff
@@ -0,0 +1,122 @@
+from paddlenlp.trainer import set_seed as seed_everything
+from ppdiffusers import ControlNetModel, StableDiffusionControlNetPipeline
+
+controlnet = ControlNetModel.from_pretrained("F:/Paddle/model/control_v11e_sd15_ip2p")
```
使用 https://huggingface.co/lllyasviel/control_v11e_sd15_ip2p 的名字 "lllyasviel/control_v11e_sd15_ip2p"

JunnYu commented:
@Submerge-Gu 很感谢你的贡献，我这里给你留了几个comment，麻烦改一下。还有的话能不能像这里，添加一下对应的文档介绍呢？
<img width="999" alt="image" src="https://github.com/PaddlePaddle/PaddleNLP/assets/50394665/458d0efb-6e81-4735-bc7e-af336d0d2ce5">

Submerge-Gu commented:
> 

好的，我现在正在修改。

GoooHi committed:
Changes to be committed:
	modified:   ppdiffusers/examples/controlnet/README.md
	modified:   ppdiffusers/examples/controlnet/gradio_ip2p2image.py
	modified:   ppdiffusers/examples/controlnet/gradio_mlsd2image.py

GoooHi commented:
@JunnYu 您好，请问我这个失败是由于啥原因呢？

JunnYu commented on ppdiffusers/examples/controlnet/README.md:
```diff
@@ -68,6 +68,20 @@ python gradio_hough2image.py
+![image](https://github.com/Submerge-Gu/Images/raw/main/4.png)
+
+## MLSD to Image
+(ControlNet V1.1) 在原基础上进行升级，采用HoughLine检测图片作为控制条件
```
"HoughLine" 这个需要纠正一下，这个是MLSD的条件。

GoooHi replied:
您好，我一开始是认为没有MLSD，后来发现之前的HoughLine就是，实际上我只是更新了下权重，需要将这块整体删掉吗

JunnYu commented on ppdiffusers/examples/controlnet/README.md:
```diff
@@ -193,3 +207,5 @@ img.save("demo.png")
 - https://github.com/lllyasviel/ControlNet/edit/main/docs/train.md
 - https://github.com/huggingface/diffusers
+
+[https://github.com/Submerge-Gu/Images/blob/main/8.png]: https://github.com/Submerge-Gu/Images/raw/main/8.png
```
这里应该删掉，我看也没用到

GoooHi committed:
Changes to be committed:
	modified:   ppdiffusers/examples/controlnet/README.md
	new file:   ppdiffusers/examples/controlnet/annotator/shuffle/__init__.py
	modified:   ppdiffusers/examples/controlnet/annotator/util.py
	renamed:    ppdiffusers/examples/controlnet/gradio_mlsd2image.py -> ppdiffusers/examples/controlnet/gradio_shuffle2image.py

JunnYu commented:
```python
model_name_list = [
    # "lllyasviel/control_v11u_sd15_tile", 存在问题
    "https://huggingface.co/lllyasviel/control_v11e_sd15_ip2p",
    "https://huggingface.co/lllyasviel/control_v11f1e_sd15_tile",
    "https://huggingface.co/lllyasviel/control_v11e_sd15_shuffle",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_softedge",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_scribble",
    "https://huggingface.co/lllyasviel/control_v11p_sd15s2_lineart_anime",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_lineart",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_seg",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_normalbae",
    "https://huggingface.co/lllyasviel/control_v11f1p_sd15_depth",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_mlsd",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_inpaint",
    "https://huggingface.co/lllyasviel/control_v11p_sd15_canny",
]
```
当前我已经转换了这么多权重，你看看有没有什么新的 anatator 部分也可以加进来？

GoooHi commented:
> ```python
> model_name_list = [
>     # "lllyasviel/control_v11u_sd15_tile", 存在问题
>     "https://huggingface.co/lllyasviel/control_v11e_sd15_ip2p",
>     "https://huggingface.co/lllyasviel/control_v11f1e_sd15_tile",
>     "https://huggingface.co/lllyasviel/control_v11e_sd15_shuffle",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_softedge",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_scribble",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15s2_lineart_anime",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_lineart",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_seg",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_normalbae",
>     "https://huggingface.co/lllyasviel/control_v11f1p_sd15_depth",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_mlsd",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_inpaint",
>     "https://huggingface.co/lllyasviel/control_v11p_sd15_canny",
> ]
> ```
> 
> 当前我已经转换了这么多权重，你看看有没有什么新的 anatator 部分也可以加进来？

@JunnYu anatator中需要转换的权重有：
annotator/lineart中的sk_model.pth和sk_model2.pth；
annotator/lineart_anime中的netG.pth;
annotator/pidinet中的table5_pidinet.pth；
我对这几个模型不是很了解，还在摸索怎么尝试转换。您看下呢？

JunnYu commented:
我看这个模型很简单的，你可以试着转成Paddle？https://github.com/lllyasviel/ControlNet-v1-1-nightly/blob/main/annotator/lineart/__init__.py
这个也可以。
https://github.com/lllyasviel/ControlNet-v1-1-nightly/blob/main/annotator/lineart_anime/__init__.py

GoooHi commented:
> 我看这个模型很简单的，你可以试着转成Paddle？https://github.com/lllyasviel/ControlNet-v1-1-nightly/blob/main/annotator/lineart/__init__.py 这个也可以。 https://github.com/lllyasviel/ControlNet-v1-1-nightly/blob/main/annotator/lineart_anime/__init__.py

@JunnYu 我现在正在尝试转成Paddle。之前没有类似的开发经验，目前现有的可以参考的资料也比较少。

JunnYu reviewed (approved):
lgtm

JunnYu commented:
你好，当前的PR我就先合入了，之后的新功能你可以重新开启一个新PR~
